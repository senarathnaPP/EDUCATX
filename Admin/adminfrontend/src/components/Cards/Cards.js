import React from 'react';
import './Cards.css';
import CardItem from './CardItem';
import user from "../../Images/staff.jpg"
import sub from "../../Images/submi.jpg"
import panel from "../../Images/panel.jpg"
import marking from "../../Images/marking.jpg"
import doc from "../../Images/doc.jpg"
import roles from "../../Images/roles.jpg"
import student from "../../Images/stud.jpg"
import research from "../../Images/research.jpg"

function Cards() {
  return (
    <div data-testid="cards" className='cards'>
      <h1>DASHBOARD</h1>
      <br/>
      <div className='cards__container'>
        <div className='cards__wrapper'>
          <ul className='cards__items'>
            <CardItem 
              src={user}
              text='Staff Management'
              label='Staff Management'
              path='/viewRoles'
            />
            <CardItem
              src={student}
              text='Student Management'
              label='Student Management'
              path='/getStudentGroups'
            />
            <CardItem
              src={panel}
              text='Allocate panel members to student groups'
              label='Panel Members Allocation'
              path='/viewPanels'
            />
            <CardItem
              src={sub}
              text='Create Submission Types'
              label='Submission Types'
              path='/viewSubmissions'
            />
          </ul>
          <ul className='cards__items'>
            <CardItem
                src={research}
                text='Research Topics'
                label='Research Topics'
                path='/getTopics'
              />
            <CardItem
              src={marking}
              text='Create Marking Schemes'
              label='Marking schemes'
              path='/viewMarkings'
            />
            <CardItem
              src={doc}
              text='Upload document/presentation templates'
              label='Document/Presentation Templates'
              path='/viewTemplates'
            />
            <CardItem
              src={roles}
              text='View Roles'
              label='View Roles'
              path='/listRoles'
            />
          </ul>
        </div>
      </div>
    </div>
  );
}

export default Cards;
