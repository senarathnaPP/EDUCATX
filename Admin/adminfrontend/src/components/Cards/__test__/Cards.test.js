import React from 'react'; 
import ReactDOM from 'react-dom';
import Cards from './../Cards';
import { BrowserRouter } from 'react-router-dom';

import { render, cleanup } from '@testing-library/react';
import "@testing-library/jest-dom/extend-expect";


import renderer from 'react-test-renderer';

afterEach(cleanup);
it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(
    <BrowserRouter>
     <Cards></Cards>
     </BrowserRouter>, div)
})

it("renders Cards correctly", () => {
    const {getByTestId} = render(<BrowserRouter><Cards><div><h1>"DASHBOARD"</h1></div></Cards></BrowserRouter>)
    expect(getByTestId('cards')).toHaveTextContent("DASHBOARD")
})

// it("renders Cards correctly", () => {
//     const {getByTestId} = render(<BrowserRouter><Cards><div><h1>"DISPLAY DASHBOARD"</h1></div></Cards></BrowserRouter>)
//     expect(getByTestId('cards')).toHaveTextContent("DASHBOARD")
// })

//This will convert this to like a virtual DOM object
it("matches snapshot", () =>{
    const tree= renderer.create(<BrowserRouter><Cards><div><h1>"DISPLAY DASHBOARD"</h1></div></Cards></BrowserRouter>).toJSON();
    expect(tree).toMatchSnapshot(); 
    //It looks for folder called snapshot
} )