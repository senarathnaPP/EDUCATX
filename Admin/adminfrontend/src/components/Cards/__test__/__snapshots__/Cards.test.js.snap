// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="cards"
  data-testid="cards"
>
  <h1>
    DASHBOARD
  </h1>
  <br />
  <div
    className="cards__container"
  >
    <div
      className="cards__wrapper"
    >
      <ul
        className="cards__items"
      >
        <li
          className="cards__item"
        >
          <a
            className="cards__item__link"
            href="/viewRoles"
            onClick={[Function]}
          >
            <figure
              className="cards__item__pic-wrap"
              data-category="Staff Management"
            >
              <img
                alt="Image"
                className="cards__item__img"
                src="staff.jpg"
              />
            </figure>
            <div
              className="cards__item__info"
            >
              <h5
                className="cards__item__text"
              >
                Staff Management
              </h5>
            </div>
          </a>
        </li>
        <li
          className="cards__item"
        >
          <a
            className="cards__item__link"
            href="/getStudentGroups"
            onClick={[Function]}
          >
            <figure
              className="cards__item__pic-wrap"
              data-category="Student Management"
            >
              <img
                alt="Image"
                className="cards__item__img"
                src="stud.jpg"
              />
            </figure>
            <div
              className="cards__item__info"
            >
              <h5
                className="cards__item__text"
              >
                Student Management
              </h5>
            </div>
          </a>
        </li>
        <li
          className="cards__item"
        >
          <a
            className="cards__item__link"
            href="/viewPanels"
            onClick={[Function]}
          >
            <figure
              className="cards__item__pic-wrap"
              data-category="Panel Members Allocation"
            >
              <img
                alt="Image"
                className="cards__item__img"
                src="panel.jpg"
              />
            </figure>
            <div
              className="cards__item__info"
            >
              <h5
                className="cards__item__text"
              >
                Allocate panel members to student groups
              </h5>
            </div>
          </a>
        </li>
        <li
          className="cards__item"
        >
          <a
            className="cards__item__link"
            href="/viewSubmissions"
            onClick={[Function]}
          >
            <figure
              className="cards__item__pic-wrap"
              data-category="Submission Types"
            >
              <img
                alt="Image"
                className="cards__item__img"
                src="submi.jpg"
              />
            </figure>
            <div
              className="cards__item__info"
            >
              <h5
                className="cards__item__text"
              >
                Create Submission Types
              </h5>
            </div>
          </a>
        </li>
      </ul>
      <ul
        className="cards__items"
      >
        <li
          className="cards__item"
        >
          <a
            className="cards__item__link"
            href="/getTopics"
            onClick={[Function]}
          >
            <figure
              className="cards__item__pic-wrap"
              data-category="Research Topics"
            >
              <img
                alt="Image"
                className="cards__item__img"
                src="research.jpg"
              />
            </figure>
            <div
              className="cards__item__info"
            >
              <h5
                className="cards__item__text"
              >
                Research Topics
              </h5>
            </div>
          </a>
        </li>
        <li
          className="cards__item"
        >
          <a
            className="cards__item__link"
            href="/viewMarkings"
            onClick={[Function]}
          >
            <figure
              className="cards__item__pic-wrap"
              data-category="Marking schemes"
            >
              <img
                alt="Image"
                className="cards__item__img"
                src="marking.jpg"
              />
            </figure>
            <div
              className="cards__item__info"
            >
              <h5
                className="cards__item__text"
              >
                Create Marking Schemes
              </h5>
            </div>
          </a>
        </li>
        <li
          className="cards__item"
        >
          <a
            className="cards__item__link"
            href="/viewTemplates"
            onClick={[Function]}
          >
            <figure
              className="cards__item__pic-wrap"
              data-category="Document/Presentation Templates"
            >
              <img
                alt="Image"
                className="cards__item__img"
                src="doc.jpg"
              />
            </figure>
            <div
              className="cards__item__info"
            >
              <h5
                className="cards__item__text"
              >
                Upload document/presentation templates
              </h5>
            </div>
          </a>
        </li>
        <li
          className="cards__item"
        >
          <a
            className="cards__item__link"
            href="/listRoles"
            onClick={[Function]}
          >
            <figure
              className="cards__item__pic-wrap"
              data-category="View Roles"
            >
              <img
                alt="Image"
                className="cards__item__img"
                src="roles.jpg"
              />
            </figure>
            <div
              className="cards__item__info"
            >
              <h5
                className="cards__item__text"
              >
                View Roles
              </h5>
            </div>
          </a>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
